cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -593,14 +593,15 @@ private Future<ConnectorStatusAndConditions> autoRestartConnector(Reconciliation
     }
 
     /**
-     * Connector or tasks failed should only restart at minute 0, 2, 6, 12, 20, 30 and 42. This method checks if there
-     * is time for another restart.
+     * Checks whether it is time for another connector restart. The decision is made based on the timing provided by
+     * {@code autoRestartBackOffInterval} and based on the {@code maxRestarts} parameter.
      *
      * @param autoRestartStatus     Status field with auto-restart status
+     * @param maxRestarts           Maximal number of restarts (or null for unlimited restarts)
```

Comment:
I think 'maximum' would be more natural English, (FTR there's an interesting discussion here about the differences https://english.stackexchange.com/questions/38260/maximum-vs-maximal).

```suggestion
     * @param maxRestarts           Maximum number of restarts (or null for unlimited restarts)
```